docs/rules/terraform_module_semver_source.md

```diff
@@ -0,0 +1,53 @@
+# terraform_module_semver_source
+
+Disallow specifying a Git or Mercurial repository as a module source without pinning to a semantic version reference.
+
+## Example
+
+```hcl
+module "unpinned" {
+  source = "git://hashicorp.com/consul.git"
+}
+
+module "git_commit" {
+  source = "git://hashicorp.com/consul.git?ref=8afd21a"
+}
+
+module "mercurial_rev" {
+  source = "hg::http://hashicorp.com/consul.hg?rev=branch"
+}
```

Comment:
It would be nice to add an example like the following:

```hcl
module "pinned_git" {
  source = "github.com/hashicorp/consul.git?ref=v1.2.3"
}
```

Reply:
Damnit, thought I’d added those, they must have got lost! Will fix.

Reply:
Ah, I had it in the example file I was testing with, just forgot to add it to the docs. 😅